pkg/controller/chaosengine/chaosengine_controller.go

```diff
@@ -644,9 +676,14 @@ func (r *ReconcileChaosEngine) removeChaosResources(engine *chaosTypes.EngineInf
 	}
 	return reconcile.Result{}, nil
 }
+
 func (r *ReconcileChaosEngine) addFinalzerToEngine(engine *chaosTypes.EngineInfo, finalizer string) error {
 	optsUpdate := client.UpdateOptions{}
-	engine.Instance.ObjectMeta.Finalizers = append(engine.Instance.ObjectMeta.Finalizers, finalizer)
+	if engine.Instance.ObjectMeta.Finalizers == nil {
+		engine.Instance.ObjectMeta.Finalizers = append(engine.Instance.ObjectMeta.Finalizers, finalizer)
+		r.recorder.Eventf(engine.Instance, corev1.EventTypeNormal, "Started reconcile for chaosEngine", "Will create all Chaos resources for chaosUID: %v", string(engine.Instance.UID))
```

Comment:
I don't think UID is required to print in log name is enough

Reply:
Also this function is related to add a finalizer then why are we logging it here?

Reply:
This isn't logging, but `describe events`

Reply:
We are doing it here, to avoid hot looping event and finalizer creation

Reply:
okay!! Then is UID required to print in the event?

Reply:
considering it appears as an event under the specified engine (describe) its not necessary to print UID. Having said that if we are sending events outside may make sense. There too name and ns should do.